We should double check some of the CI failures since they are in `signal`:

```
...
 FAILED scipy/signal/tests/test_signaltools.py::TestCorrelateReal::test_rank3_valid[uint8] - OverflowError: Python integer 504 out of bounds for uint8
FAILED scipy/signal/tests/test_signaltools.py::TestCorrelateReal::test_rank3_valid[int8] - OverflowError: Python integer 184 out of bounds for int8
FAILED scipy/signal/tests/test_signaltools.py::TestCorrelateReal::test_rank3_same[uint8] - OverflowError: Python integer 504 out of bounds for uint8
...
```

--- a/scipy/signal/_spectral_py.py
+++ b/scipy/signal/_spectral_py.py
@@ -1987,16 +1987,15 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
 
     .. versionadded:: 0.16.0
     """
-    # Created strided array of data segments
+    # Created sliding window view of array
     if nperseg == 1 and noverlap == 0:
         result = x[..., np.newaxis]
     else:
-        # https://stackoverflow.com/a/5568169
         step = nperseg - noverlap
-        shape = x.shape[:-1]+((x.shape[-1]-noverlap)//step, nperseg)
-        strides = x.strides[:-1]+(step*x.strides[-1], x.strides[-1])
-        result = np.lib.stride_tricks.as_strided(x, shape=shape,
-                                                 strides=strides)
+        result = np.lib.stride_tricks.sliding_window_view(
+            x, window_shape=nperseg, axis=-1, writeable=True
+        )
+        result = result[..., 0::step, :]
 
     # Detrend each data segment individually
     result = detrend_func(result)

